src/main/java/com/notably/model/BlockModel.java

```diff
@@ -0,0 +1,53 @@
+package com.notably.model;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.model.block.Block;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeItem;
+
+/**
+ * API of the BlockModel component.
+ */
+public interface BlockModel {
+    /**
+     * Gets the BlockTree currently in use by Notably.
+     */
+    BlockTree getBlockTree();
+
+    /**
+     * Check if currently open block has the specified child block.
+     */
+    boolean hasChild(Block b);
+
+    /**
+     * Adds a block to the specified path, if possible.
+     */
+    void addBlock(AbsolutePath p, Block b);
+
+    /**
+     * Removes the block at the specified path, if possible.
+     */
+    void removeBlock(AbsolutePath p);
+
+    /**
+     * Opens the block at the specified path, if possible.
+     */
+    BlockTreeItem openBlock(AbsolutePath p);
```

Comment:
Actually, shouldn't this method be `void` too? Whenever the open path changes, the `currentlyOpenPathProperty` below should trigger the change, and the UI will react accordingly.

Reply:
Sure, makes sense!